digitalforms-api/src/test/java/ca/bc/gov/open/digitalformsapi/viirp/controller/DocumentsApiControllerTests.java

```diff
@@ -323,4 +361,133 @@ public void documentsAssociationNoticeDocumentIdPostMissingNoticeTypeCdFailure()
+        GetDocumentsListServiceResponse result = controllerResponse.getBody();
+        Assertions.assertTrue(result.getResults().size() > 0);
+        Assertions.assertEquals("MV2702A", result.getResults().get(0).getDocumentTypeCd());
+    }
+    
+    @DisplayName("GET, Documents Meta List with Prohibition Id Success - Documents API Delegate")  
+    @Test
+	public void testDocumentsApiGETDocumentsWithProhibitionIdSuccess() throws Exception {
+    	
+    	String correlationId = DigitalFormsConstants.UNIT_TEST_CORRELATION_ID;
+    	String noticeNo = DigitalFormsConstants.UNIT_TEST_NOTICE_NUMBER;
+    	Long prohibitionId = DigitalFormsConstants.UNIT_TEST_PROHIBITION_ID;
+    	
+    	// Mock underlying VIPS REST Impoundment Search Failure
+		vipsSearch.setRespMsg(DigitalFormsConstants.VIPS_NOTICE_NOT_FOUND);
+		vipsSearch.setRespCd(DigitalFormsConstants.VIPSWS_GENERAL_FAILURE_CD);
+        Mockito.when(service.searchImpoundment(correlationId, noticeNo)).thenReturn(vipsSearch);
+		
+		// Mock underlying VIPS REST Prohibition Search
+		vipsProhibitionSearch.setRespMsg(DigitalFormsConstants.DIGITALFORMS_SUCCESS_MSG);
+		vipsProhibitionSearch.setRespCd(DigitalFormsConstants.VIPSWS_SUCCESS_CD);
+        Mockito.when(service.searchProhibition(correlationId, noticeNo)).thenReturn(vipsProhibitionSearch);
+        
+        // Mock underlying VIPS REST Get Documents response in good case 
+        vipsDocumentsResponse.setRespMsg(DigitalFormsConstants.DIGITALFORMS_SUCCESS_MSG);
+        vipsDocumentsResponse.setRespCd(DigitalFormsConstants.VIPSWS_SUCCESS_CD);
+        Mockito.when(service.getDocumentsMetaList(correlationId, null, prohibitionId)).thenReturn(vipsDocumentsResponse);
+        
+        // Create successful get documents call with noticeNo and validate response 
+        ResponseEntity<GetDocumentsListServiceResponse> controllerResponse = controller.documentsListNoticeNoCorrelationIdGet(noticeNo, correlationId);
+        GetDocumentsListServiceResponse result = controllerResponse.getBody();
+        Assertions.assertTrue(result.getResults().size() > 0);
+        Assertions.assertEquals("MV2702A", result.getResults().get(0).getDocumentTypeCd());
+    }
+    
+    @DisplayName("GET, Documents Not Found - Documents API Delegate")  
+    @Test
+	public void testDocumentsApiGETDocumentsNotFound() throws Exception {
+    	
+    	String correlationId = DigitalFormsConstants.UNIT_TEST_CORRELATION_ID;
+    	String noticeNo = DigitalFormsConstants.UNIT_TEST_NOTICE_NUMBER;
+    	Long impoundmentId = DigitalFormsConstants.UNIT_TEST_IMPOUNDMENT_ID;
+		
+		// Mock underlying VIPS REST Search 
+		vipsSearch.setRespMsg(DigitalFormsConstants.DIGITALFORMS_SUCCESS_MSG);
+		vipsSearch.setRespCd(DigitalFormsConstants.VIPSWS_SUCCESS_CD);
+        Mockito.when(service.searchImpoundment(correlationId, noticeNo)).thenReturn(vipsSearch);
+        
+        // Mock underlying VIPS REST Get Documents response returning no documents in the response list even though the
```

Comment:
Hi Burak. This comment is confusing. A list of document may not exist even if we have a impoundment Id or prohibition Id returned from the prior search. This is not a BAD vips case. Maybe just remove the comment. Thanks.

Reply:
Sorry that's a copy paste error. I will remove that part of the comment.